pkgs/applications/editors/vim/macvim.nix

```diff
@@ -6,14 +6,24 @@
 , gettext
 , pkg-config
 , cscope
-, ruby
+, ruby_3_2
 , tcl
-, perl
+, perl536
 , luajit
 , darwin
-, python3
+, python311
 }:
 
+# Try to match MacVim's documented script interface compatibility
+let
+  # Perl 5.30 - closest we get is 5.36. 5.38 is currently failing
+  perl = perl536;
+  # Python3 3.9 or above - the binary release builds against 3.11 so we will too
+  python3 = python311;
```

Comment:
python3 defaults to that, do we really need to hardcode it here? Same for ruby_3_2.

Reply:
At the time this PR was written, `python3` evaluated to `python310` instead. And `ruby` still evaluates to `ruby_3_1`. I'm open to changing this to just `python3` as MacVim documents compatibility with "Python3 3.9 or above", but the ruby support is documented as "Ruby 3.2" so we should match that.

Reply:
I've removed the `python311` pin but left `ruby_3_2`.